cmd/ipi-deprovision/ipi-deprovision.sh

```diff
@@ -1,17 +1,38 @@
-#!/bin/sh
+#!/bin/bash
 
-### expirationDate is always with timezone +00:00, eg, "2019-07-23T22:35+0000"
-MY_DATE=$(TZ=":Africa/Abidjan" date '+%Y-%m-%d')
-declare -a regions=("us-east-1" "us-east-2" "us-west-1")
+set -o errexit
+set -o nounset
+set -o pipefail
+
+### cluster_age_cutoff is going to be used as argument in the aws command below
+### to filter out the expired clusters with the tags of the key expirationDate
+### which is always with timezone +00:00, eg, "2019-07-23T22:35+0000"
+cluster_age_cutoff=$(TZ=":Africa/Abidjan" date '+%Y-%m-%d')
+
+if [[ -n "${CLUSTER_TTL-}" ]]; then
+  ### eg, export CLUSTER_TTL="22 hours ago"
+  echo "Searching for clusters with a TTL of ${CLUSTER_TTL}"
+  cluster_age_cutoff="$(TZ=":Africa/Abidjan" date --date="${CLUSTER_TTL}" '+%Y-%m-%dT%H:%M+0000')"
+fi
+
+echo "cluster_age_cutoff: ${cluster_age_cutoff}"
+
+### eg, export AWS_REGIONS="us-east-1;us-west-1"
+echo "take regions from env. var.: AWS_REGIONS: ${AWS_REGIONS}"
+IFS=';' read -r -a regions <<< "$AWS_REGIONS"
 
 handle_cluster () {
   local cluster_name;
   cluster_name=$1
-  echo "handling cluster: ${cluster_name} ..."
+  echo "handling cluster: ${cluster_name} with expirationDate: $(aws ec2 describe-vpcs --output json --region "us-east-1" --filters "Name=tag:${cluster_name},Values=owned" | jq -r '.Vpcs[].Tags[] | select (.Key=="expirationDate") | .Value')"
```

Comment:
nit: this will make runtime much slower as we now make O(N) aws calls (rather than one list) and be also more costly on API use -- we can save the output of `aws describe-vpcs --output json` into a file and post-process that or use `jq` in the first call to select the cluster name and expiration date, either way would work.

Reply:
Done.
I like the 2nd way more but i can implement the 1st one for the moment.